server/src/main/java/io/crate/protocols/http/Headers.java

```diff
@@ -66,25 +66,55 @@ public static void setKeepAlive(HttpVersion httpVersion, FullHttpResponse resp)
         }
     }
 
-    public static Tuple<String, SecureString> extractCredentialsFromHttpBasicAuthHeader(String authHeaderValue) {
+    /**
+     * @return tuple where first entry is a role name
+     * and second entry is either passord or token, depending on auth method
+     */
+    public static Tuple<String, SecureString> extractCredentialsFromHttpAuthHeader(String authHeaderValue) {
```

Comment:
I highly suggest to keep the existing method for the `Basic` auth method and create a dedicated one for the token. This new code looks like is doesn't share much but all logic is split into 2 if/else branches.

Reply:
yep, had the same thoughts on typing this, it's just a "dirty draft"